File: backend/src/models/acidente.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class AcidenteCreate(BaseModel):
    latitude: float
    longitude: float
    endereco: Optional[str] = None
    descricao: str
    gravidade: str = "MODERADO"
    tipo_acidente: str = "OUTRO"
    causa_principal: Optional[str] = None
    numero_vitimas: int = 0
    numero_veiculos: int = 1
    origem_registro: str = "WEB_POLICIA"
    fotos: Optional[List[str]] = []


class AcidenteUpdate(BaseModel):
    status: Optional[str] = None
    gravidade: Optional[str] = None
    descricao: Optional[str] = None
    confirmado_oficialmente: Optional[bool] = None
    causa_principal: Optional[str] = None


class AcidenteResponse(BaseModel):
    acidente_id: str
    latitude: float
    longitude: float
    endereco: Optional[str] = None
    descricao: str
    gravidade: str
    tipo_acidente: str
    causa_principal: Optional[str] = None
    numero_vitimas: int
    numero_veiculos: int
    status: str
    origem_registro: str
    confirmado_oficialmente: bool
    fotos: List[str]
    created_at: datetime
    updated_at: datetime
    created_by: Optional[str] = None

File: backend/src/models/Acidente.js
const mongoose = require('mongoose');

const acidenteSchema = new mongoose.Schema({
  acidente_id: { type: String },
  latitude: { type: Number, required: true },
  longitude: { type: Number, required: true },
  endereco: { type: String, default: null },
  descricao: { type: String, default: '' },
  gravidade: { type: String, enum: ['GRAVE', 'MODERADO', 'LEVE'], default: 'MODERADO' },
  tipo_acidente: { type: String, default: 'OUTRO' },
  causa_principal: { type: String, default: '' },
  numero_vitimas: { type: Number, default: 0 },
  numero_veiculos: { type: Number, default: 0 },
  status: { type: String, enum: ['REPORTADO', 'VALIDADO', 'EM_ATENDIMENTO', 'RESOLVIDO'], default: 'REPORTADO' },
  origem_registro: { type: String, default: 'WEB_POLICIA' },
  confirmado_oficialmente: { type: Boolean, default: false },
  fotos: [{ type: String }],
  created_at: { type: Date, default: Date.now },
  updated_at: { type: Date, default: Date.now },
  created_by: { type: String, default: '' },
  updated_by: { type: String, default: '' }
}, { collection: 'acidentes', timestamps: false });

module.exports = mongoose.model('Acidente', acidenteSchema);
